tests/regression/kernel/test_callstack

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+#
+# Copyright (C) - 2017 Francis Deslauriers <francis.deslauriers@efficios.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License, version 2 only, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+TEST_DESC="Kernel tracer - Callstack context"
+
+CURDIR=$(dirname $0)/
+TESTDIR=$CURDIR/../..
+NUM_TESTS=18
+TEST_APP="$TESTDIR/utils/testapp/gen-syscall-events-callstack/gen-syscall-events-callstack"
+PARSE_CALLSTACK="$TESTDIR/utils/parse-callstack.py"
+
+SESSION_NAME="callstack"
+CHANNEL_NAME="chan0"
+
+source $TESTDIR/utils/utils.sh
+
+function lttng_untrack_all()
+{
+	lttng untrack -s "$SESSION_NAME" --all --pid -k > /dev/null
+}
+
+function lttng_track_pid()
+{
+	local PID=$1
+	lttng track -s "$SESSION_NAME" -k --pid="$PID" > /dev/null
+}
+
+function run_workload()
+{
+	local file_sync=$(mktemp -u)
+
+	lttng_untrack_all
+
+	./"$TEST_APP" $file_sync &
+	PID=$!
+	lttng_track_pid $PID
+
+	# Wait for the testapp to create the synchronization file
+	while [ ! -f "$file_sync" ]; do
+		sleep 0.5
+	done
+
+	start_lttng_tracing_ok
+
+	# Send signal to trigger the syscall
+	kill -SIGUSR1 $PID
```

Comment:
Ah, yes. Agreed that it works. I would be tempted to stay clear from signals here since it's straightforward to do. Can you use the same scheme as tests/utils/testapp/gen-ust-events ? See mainly create_file() and wait_on_file(). We should perhaps move those to a helper library for test applications. This way, both rendez-vous points (in each direction) end up being file-based.